src/gui/editors/TimeLineWidget.cpp

```diff
@@ -225,83 +197,139 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
 		}
 	}
 
-	// Draw the main rectangle (outer border)
+	// Draw the loop rectangle's outer outline
 	p.setPen( loopPointsActive ? getActiveLoopColor() : getInactiveLoopColor() );
 	p.setBrush( Qt::NoBrush );
 	p.drawRect( outerRectangle );
 
-	// Draw the inner border outline (no fill)
+	// Draw the loop rectangle's inner outline
 	QRect innerRectangle = outerRectangle.adjusted( 1, 1, -1, -1 );
 	p.setPen( loopPointsActive ? getActiveLoopInnerColor() : getInactiveLoopInnerColor() );
 	p.setBrush( Qt::NoBrush );
 	p.drawRect( innerRectangle );
+	
+	// Draw loop handles if necessary
+	const auto handleMode = ConfigManager::inst()->value("app", "loopmarkermode") == "handles";
+	if (handleMode && underMouse() && QGuiApplication::keyboardModifiers().testFlag(Qt::ShiftModifier))
+	{
+		const auto handleWidth = std::min(m_loopHandleWidth, loopRectWidth / 2 - 1);
+		const auto leftHandle = QRectF(loopStart - .5, loopRectMargin - .5, handleWidth, loopRectHeight);
+		const auto rightHandle = QRectF(loopEndR - handleWidth - .5, loopRectMargin - .5, handleWidth, loopRectHeight);
+		const auto color = loopPointsActive ? m_activeLoopHandleColor : m_inactiveLoopHandleColor;
+
+		p.fillRect(leftHandle, color);
+		p.fillRect(rightHandle, color);
+	}
 
 	// Only draw the position marker if the position line is in view
-	if (m_posMarkerX >= m_xOffset && m_posMarkerX < width() - m_posMarkerPixmap.width() / 2)
+	if (markerX(m_pos) >= m_xOffset && markerX(m_pos) < width() - m_posMarkerPixmap.width() / 2)
 	{
 		// Let the position marker extrude to the left
 		p.setClipping(false);
 		p.setOpacity(0.6);
-		p.drawPixmap(m_posMarkerX, height() - m_posMarkerPixmap.height(), m_posMarkerPixmap);
+		p.drawPixmap(markerX(m_pos) - (m_posMarkerPixmap.width() / 2),
+			height() - m_posMarkerPixmap.height(), m_posMarkerPixmap);
 	}
 }
 
+auto TimeLineWidget::getClickedTime(const int xPosition) const -> TimePos
```

Comment:
Having the `const` in `const int xPosition` here is redundant.

Reply:
Okay, so I happened to be right here. It's not that its inherently wrong to make `xPosition` `const`, [but its overly pedantic since function arguments that are passed by value are rarely ever mutated](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#S-const).

I didn't mention it because I actually didn't know about this for sure (just went off a feeling that this was off but couldn't tell why), so I had to dig through the guidelines. No action needed, but good to know in the future.

Reply:
My understanding of their comment is that code analysis tools should not emit a warning if const is omitted from function parameters, since this will likely produce a large number of warnings for little benefit. I don't think they mean that const is actually better omitted in those cases.

Reply:
Yeah, "redundant" wasn't the best way to put it, because if it was, analysis tools would probably pick up that as a warning. I meant it more as in that it can confuse more than help.

Reply:
`Con.1` just says don't enforce the rule of const by default for function parameters.
```
Function parameters passed by value are rarely mutated,
but also rarely declared const. To avoid confusion and lots
of false positives, don’t enforce this rule for function
parameters.

    void g(const int i) { ... }  // pedantic

Note that a function parameter is a local variable so
changes to it are local.
```
The comment says pedantic, but that is just an opinion honestly. Nothing about this says we shouldn't declare function parameters const.

What we are doing though is self-documentation and declaring to anyone looking that `xPosition` is a value that shall not be changed within the function. Like it says, if something should not change, declare it `const`. For such a large codebase in my opinion it's better to be explicit.

Reply:
Fair, I do like the mindset of "const by default, mutable when you need to". Parameters passed by value can also be seen local to a function, so if we already agree that we should declare local variables `const` that are within the body, they can be applied to function parameters too since the scope is the same.

Did more looking into this. Some say its useful, some say it isn't. It's quite common for me to not have strong opinions on these kind of things, usually leading me to consider a credible source (i.e., the guidelines) as a source of truth and absolute correctness. But then again, I shouldn't really treat it as such, and actually like a guideline because not everything in there I fully agree with anyways (I just feel like some of them miss cruical details).

It just so happened that I did not really have a strong opinion for this, but I agree personally that the communication between the API and its users becomes clearer using `const`. In any case though, its definitely not something worth discussing over in many cases in terms of enforcing it or not, but it did have me thinking.

Reply:
Conversations like this are always useful though. 😄